src/rabbit_reader.erl

```diff
@@ -759,6 +743,30 @@ maybe_close(State) ->
 termination_kind(normal) -> controlled;
 termination_kind(_)      -> uncontrolled.
 
+format_hard_error(#amqp_error{name = N, explanation = E, method = M}) ->
+    io_lib:format("operation ~s caused a connection exception ~s: ~p", [M, N, E]);
+format_hard_error(Reason) ->
+    Reason.
+
+log_connection_exception(Name, Ex) ->
+  Severity = case Ex of
```

Comment:
since you are refactoring, why not reformat the case here? either move it to a new line or add the Emacs spacing?